gui/common_qt_functions.py

```diff
@@ -69,21 +78,30 @@ def setAzureSettings(self):
             'auth-resource': '',
             'update-md5': False,
             'validate-md5': False,
-            'virtual-directory': False,
-            'disable-compression': False
+            'virtual-directory': True,
+            'disable-compression': False,
+            'max-results-for-list': 2,
```

Comment:
I think we talked about this already, so I wanted to add this in the comment for future reference. These new settings will not be expressed by the GUI until the .ui code is modified to show these. I am fine using the code here to enforce a yaml schema, we will just need to note what changed in this PR to then update the ui code later.

Reply:
I'll add some comments here to indicate the options that are not exposed in the GUI itself.
I think it's okay to leave many options invisible to GUI users. Users can always edit the config file directly.